### DISCUSSION TITLE
General understanding and wording

### DISCUSSION THREAD

**Discussion by RealZendor** [PinataCloud/docs#45] [General]
Hi guys, although I've been developing web applications for more than 20 years, I'm looking like an idiot. I don't understand the basics and unfortunately they are not explained in the documentation. These are my main questions:
1. is there a difference between "uploading a file" and "pinning a file"?
2. if so, what does "pinning" a file mean?
3. i want to use PinataCloud exclusively as a cloud storage to store files and later make them publicly available in websites? What rights does the API key need to have?
4. what is IPFS?
Thanks for any advice!

**Comment by stevedylandev** [comment]
Hey there! These are actually great questions, let me help you out! Going to answer in an order that will help make the next answer more clear :) 

4. IPFS stands for the [InterPlanetary File System](https://ipfs.io), and is something Pinata has been primarily known for over the past 5 years. It's a distributed and immutable file sharing network that is often used in blockchain applications. It's similar to storage where the file has to be hosted, but the next step is making it available on the peer-to-peer network which is what we specialize in. Outside of Web3 stuff it's not used much as files can only be public, and that's why we built our Files API. 
1 & 2. "Pinning" is the term used for putting a file on IPFS and making it available on the network. Uploading is more standard cloud file storage. 
3. Great use case! I would actually recommend making an admin API key and using it server-side for most operations, such as [making a group of files public](https://docs.pinata.cloud/files/file-groups#updating-a-group). For uploads we have [a method for client side uploads](https://docs.pinata.cloud/files/uploading-files#user-generated-keys), however we will soon be releasing a more recognizable signed upload url flow. 

This is really helpful perspective for us so we can adjust our docs to make these distinctions a bit more clear. Thank you for asking! Please let me know if you have more questions :)
